File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.cpp
class Solution {
public:
    int countValidSelections(vector<int>& nums) {
        int n = nums.size();
        if (n == 0) return 0;
        
        int count = 0;
        vector<int> left(n, 0);
        vector<int> right(n, 0);

        for (int i = 1; i < n; i++) {
            left[i] = left[i - 1] + nums[i - 1];
        }

        for (int i = n - 2; i >= 0; i--) {
            right[i] = right[i + 1] + nums[i + 1];
        }

        for (int i = 0; i < n; i++) {
            if (nums[i] != 0) continue;

            if (left[i] == right[i])
                count += 2;
            else if (abs(left[i] - right[i]) == 1)
                count += 1;
        }

        return count;
    }
};

File: 3354-make-array-elements-equal-to-zero/3354-make-array-elements-equal-to-zero.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        n = len(nums)
        prefix = [0] * n
        suffix = [0] * n
        for i in range (1, n):
            prefix[i] += prefix[i-1] + nums[i - 1]
        for i in range (n-2, -1, -1):
            suffix[i] += suffix[i + 1] + nums[i + 1]
        cnt = 0
        for i in range (0, n):
            if nums[i] == 0:
                if prefix[i] == suffix[i]:
                    cnt+=2
                elif abs(prefix[i] - suffix[i]) == 1:
                    cnt+=1
        return cnt